src/main/java/org/folio/service/manager/export/ExportManagerImpl.java

```diff
@@ -87,48 +93,89 @@ public void exportData(JsonObject request) {
    *
    * @param exportPayload payload of the export request
    */
-  protected void exportBlocking(ExportPayload exportPayload) {
+  protected void exportBlocking(ExportPayload exportPayload, Promise blockingPromise) {
     List<String> identifiers = exportPayload.getIdentifiers();
     FileDefinition fileExportDefinition = exportPayload.getFileExportDefinition();
     MappingProfile mappingProfile = exportPayload.getMappingProfile();
     OkapiConnectionParams params = exportPayload.getOkapiConnectionParams();
-    SrsLoadResult srsLoadResult = new SrsLoadResult();
-    /**
-     * For Q3-2020, if a custom mapping profile is used, never fetch the underlying SRS records, always generate marc records on the
-     * fly
-     */
-    if (isTransformationEmpty(mappingProfile)) {
-      srsLoadResult = loadSrsMarcRecordsInPartitions(identifiers, exportPayload.getJobExecutionId(), params);
-      LOGGER.info("Records that are not present in SRS: {}", srsLoadResult.getInstanceIdsWithoutSrs());
 
+    if (Objects.nonNull(mappingProfile)
+        && mappingProfile.getRecordTypes().contains(RecordType.SRS)
+        && !mappingProfile.getRecordTypes().contains(RecordType.INSTANCE)) {
+      SrsLoadResult srsLoadResult = loadSrsMarcRecordsInPartitions(identifiers, exportPayload.getJobExecutionId(), params);
+      LOGGER.info("Records that are not present in SRS: {}", srsLoadResult.getInstanceIdsWithoutSrs());
       List<String> marcToExport = srsRecordService.transformSrsRecords(mappingProfile, srsLoadResult.getUnderlyingMarcRecords(),
         exportPayload.getJobExecutionId(), params);
       exportService.exportSrsRecord(marcToExport, fileExportDefinition);
       LOGGER.info("Number of instances not found in SRS: {}", srsLoadResult.getInstanceIdsWithoutSrs().size());
+      mappingProfileService
+          .getById(DEFAULT_MAPPING_PROFILE_ID, params.getTenantId())
+          .onSuccess(
+              defaultMappingProfile -> {
+                generateRecordsOnTheFly(
+                    exportPayload,
+                    identifiers,
+                    fileExportDefinition,
+                    defaultMappingProfile,
+                    params,
+                    srsLoadResult);
+                blockingPromise.complete();
+              })
+          .onFailure(
```

Comment:
I don't like this manner of code formatting, let's revert it back to the usual version, and discuss, it with all the backend people

Reply:
This makes the async code completely unreadable for me